src/main/java/com/insert/ogbsm/service/timeTable/TimeTableProvider.java

```diff
@@ -0,0 +1,77 @@
+package com.insert.ogbsm.service.timeTable;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import com.insert.ogbsm.domain.timeTable.TimeTable;
+import com.insert.ogbsm.presentation.timeTable.dto.RowTimeTableReq;
+import lombok.RequiredArgsConstructor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+@RequiredArgsConstructor
+public class TimeTableProvider {
+    private final OkHttpClient httpClient;
+
+    @Value("${env.neis.time_table_url}")
+    private String TIME_TABLE_API_URL;
+
+    @Nullable
+    private static List<RowTimeTableReq> getTimeTableDtos(JsonElement element) {
+        JsonArray jsonElements = element
+                .getAsJsonObject().get("hisTimetable")
+                .getAsJsonArray().get(1)
+                .getAsJsonObject().get("row")
+                .getAsJsonArray();
+
+        Type RowTimeTableList = new TypeToken<List<RowTimeTableReq>>() {
+        }.getType();
+
+        return new Gson().fromJson(jsonElements, RowTimeTableList);
+    }
+
+    public List<TimeTable> getRowWeekTimeTableList(LocalDate date) throws IOException {
+        String url = getUrl(date);
+        Response timteTableResponse = sendRequest(url);
+        JsonElement element = JsonParser.parseString(requireNonNull(timteTableResponse.body()).string());
+
+        return requireNonNull(getTimeTableDtos(element))
+                .stream()
+                .map(RowTimeTableReq::toEntity)
+                .toList();
+    }
+
+    @NotNull
+    private Response sendRequest(String url) throws IOException {
+        Request mealRequest = new Request.Builder()
+                .url(url)
+                .get()
+                .build();
+        Response mealResponse = httpClient.newCall(mealRequest).execute();
+        return mealResponse;
+    }
+
+    private String getUrl(LocalDate date) {
+        if (date.getMonthValue() < 10) {
+            return TIME_TABLE_API_URL + "ALL_TI_YMD=" + date.getYear() + "0" + date.getMonthValue() + date.getDayOfMonth();
+        }
+
+        return TIME_TABLE_API_URL + "ALL_TI_YMD=" + date.getYear() + date.getMonthValue();
+    }
```

Comment:
```suggestion
    private String getUrl(LocalDate date) {
        return TIME_TABLE_API_URL + "ALL_TI_YMD=" + date.format(DateTimeFormatter.BASIC_ISO_DATE);
    }
```

위 아래 로직이 왜 다른지는 이해 하지 못해서 정리만 해 보았습니다?

Reply:
<img width="484" alt="스크린샷 2023-10-03 오후 5 19 24" src="https://github.com/Team-INSERT/Something_Big_Backend/assets/101192100/9ca1466e-f288-49f0-af7c-c59e7a8b96e5">

년월일 중에 일을 포함하고 싶지 않기 때문입니다

Reply:
왓더 호현님께서 리뷰해주신 것 보고 조금 더 생각을 해봤는데 로직에서 "일"이 꼭 있어야 했습니다! 와우 🤩